pkg/errors/armerrors.go

```diff
@@ -34,7 +35,7 @@ func IsResponseError(err error) *azcore.ResponseError {
 // IsNotFoundErr is used to determine if we are failing to find a resource within azure.
 func IsNotFoundErr(err error) bool {
 	azErr := IsResponseError(err)
-	return azErr != nil && azErr.ErrorCode == ResourceNotFound
+	return azErr != nil && azErr.StatusCode == http.StatusNotFound
```

Comment:
This seems okay to me I think, but e.g. in Karpenter Provider we still use the `ResourceNotFound` check for certain resources, https://github.com/search?q=repo%3AAzure%2Fkarpenter-provider-azure%20ResourceNotFound&type=code

Might be a good idea to modify this to check `StatusCode` there instead too, for consistency?